modules/internal/ChapelArray.chpl

```diff
@@ -1473,6 +1483,34 @@ module ChapelArray {
     }
   }  // record _domain
 
+  proc +=(ref sd: domain, inds: [] sd.idxType) 
+    where isSparseDom(sd) && sd.rank == 1{
+    
+    sd.bulkAdd(inds, false, false);
+  }
+
+  proc +=(ref sd: domain, inds: [] sd.rank*sd.idxType ) 
+    where isSparseDom(sd) && sd.rank > 1 {
+
+    sd.bulkAdd(inds, false, false);
+  }
+
+  /*
+    Currently this is not optimized for addition of a sparse
+  */
+  proc +=(ref sd: domain, d: domain) 
+    where isSparseDom(sd) && d.rank==sd.rank && sd.idxType==d.idxType {
+
+    type _idxType = if sd.rank==1 then int else sd.rank*int;
+    const indCount = d._value.dsiNumIndices;
+    const arr: [{0..#indCount}] _idxType;
+  
+    //this could be a parallel loop. but ranks don't match -- doesn't compile
```

Comment:
I somehow couldn't find a universal way of getting all indices in a domain(including dense and sparse), and a zippered forall over 1-D array and 2-D domain didn't work obviously.